tests/testthat/test-oauth-query-params.R

```diff
@@ -0,0 +1,67 @@
+context("OAuth-query-params")
+
+test_that("oauth2.0 authorize url appends query params", {
+  app <- oauth_app("x", "y", "z")
+  scope <- httr:::check_scope(c("foo", "bar"))
+  queryParams <- list(
+    foo = "bar",
+    paul = "isCool"
+  )
+  authURL <- oauth2.0_authorize_url(
+    endpoint = oauth_endpoints("google"),
+    app = app,
+    scope = scope,
+    auth_page_query_params = queryParams
+  )
+
+  url <- parse_url(authURL)
+  paramNames <- names(queryParams)
+  for (qpIndex in 1:length(queryParams))
+  {
+    paramName <- paramNames[qpIndex]
+    expect_equal(url$query[paramName], queryParams[qpIndex])
+  }
+})
+
+test_that("oauth2.0 authorize url does not append empty query params", {
```

Comment:
What are you trying to test here? I can't tell because there's so much indirection in your code. Can you please simplify?

Reply:
Trying to test that with an empty/null query_extra parameter the url still looks like it did before my change (i.e. common elements untouched). 
Tried to simplify/clarify - it's nowhere near perfect I just can't think of a more effective way to test for this without testing for each parameter in the authorization URL.